### PR TITLE
Fix VarInt decoding with extra bytes at the end

### PR DESCRIPTION
**Pull request by jamesbornholt**
After #10, decoding a VarInt from a slice can fail if the final byte of the slice has its MSB set, even if the VarInt decoding will never make it that far. This is an issue if we're decoding from a larger slice that has other encoded data after the VarInt.

It's not enough to check the last byte of the slice for the MSB being set -- we have to scan all the bytes and try to decode them, and only fail decoding if we never saw the MSB set in any byte.

### CODE REVIEW

File: src/varint_tests.rs [hunk 1]
```diff
@@ -166,4 +166,13 @@ mod tests {
...
         assert!(read.read_varint::<u64>().is_err());
     }
+
+    #[test]
+    fn test_varint_with_extra_bytes() {
+        let mut encoded = 0x12345usize.encode_var_vec();
+        assert_eq!(usize::decode_var(&encoded[..]), Some((0x12345, 3)));
+
+        encoded.push(0x99);
+        assert_eq!(usize::decode_var(&encoded[..]), Some((0x12345, 3)));
```
Comment (dermesser) [inline src/varint_tests.rs:176]: would you mind adding another short case testing `usize::decode_var(...) == None`? I.e., if there is no byte with MSB not set. I believe the test just before this one tests it too, but with a Reader.

Thank you! LGTM otherwise.

**Review by dermesser — CHANGES_REQUESTED** [review]

File: src/varint_tests.rs (hunk 1 above)
Comment (jamesbornholt) [reply src/varint_tests.rs]: Good idea, done!

**Comment by jamesbornholt** [comment]
I realized I forgot to make the same change for `i64`, which is done now.

I also added another test for overflow (encodings that are too long to produce a valid `u64`/`i64`), which exposed an off-by-one in the conditional: `shift > (10 * 7)` should be `shift > (9 * 7)` to avoid trying to do a shift by > 64 bits.

**Review by dermesser — APPROVED** [review]
